includes/admin/stripe-settings.php

```diff
@@ -180,6 +180,18 @@
 				'long'  => __( 'Text and logo', 'woocommerce-gateway-stripe' ),
 			],
 		],
+		'payment_request_locations'           => [
+			'title'       => __( 'Payment Request Button Locations', 'woocommerce-gateway-stripe' ),
+			'type'        => 'multiselect',
+			'description' => __( 'Select where you would like Payment Request Buttons to be displayed', 'woocommerce-gateway-stripe' ),
+			'desc_tip'    => true,
+			'options'     => [
+				'product'  => __( 'Product pages', 'woocommerce-gateway-stripe' ),
+				'cart'     => __( 'Cart page', 'woocommerce-gateway-stripe' ),
+				'checkout' => __( 'Checkout page', 'woocommerce-gateway-stripe' ),
+			],
+			'default'     => [ 'product', 'cart' ],
```

Comment:
There's one issue here. This will not work until the user submits the Stripe settings form. The two options will be selected while the value is not persisted in the DB yet.

<img width="659" alt="Screen Shot 2021-08-11 at 12 04 44 AM" src="https://user-images.githubusercontent.com/1025173/128972477-5c3352de-827c-4fba-8295-e9e7fa7ee241.png">

Is this going to be saved/updated automatically when the user updates the plugin? Otherwise we will show incorrect state for this setting to the user.

Reply:
> This will not work until the user submits the Stripe settings form. The two options will be selected while the value is not persisted in the DB yet.

This will work in the sense that if there's nothing saved to the DB the default values are used, and that's intended. Nothing should (or has) to be saved to the database until the merchant updates the plugin settings.

I added some logic in e3ad0f5655125d5cc7b3ad6b9641a128ed2ba591 to retrieve the values during an update, but I don't think we should keep that logic. We're also discussing this in p1628641383050300-slack-CHG7MTCAF. I've pinged you in the thread.

Reply:
Ah, like we discussed in Slack, you were right; there was an issue with relying on the default value.

I've fixed that in e6901e7a30c5ed9500a3facf36f24711b3965d6c 🎉

Reply:
I think e3ad0f5 also fixes the issue. But not for when switching to your branch and `payment_request_location` was not defined yet. This last case is fixed by e6901e7